src/esm_runscripts/config_control.py

```diff
@@ -0,0 +1,112 @@
+    version_from_runscript = runscript[model].get("version", None)
+
+    # populate list of available versions
+    available_versions = []
+    items = model_config.get("available_versions", None)
+    if items:
+        available_versions += items
+    items = model_config.get("choose_version", None)
+    if items:
+        available_versions += list(items)
+    if available_versions:
+        # TODO: I have a better way of handling int/str in versions but this is
+        # for the clean-up phase
+        available_versions = [str(item) for item in available_versions]
+        available_versions = sorted(list(set(available_versions)))
+
+    if version_from_runscript is None:
+        return True
+    # elif "available_versions" not in model_config:
+    elif available_versions is None:
+        import ipdb; ipdb.set_trace()
+        msg = f"Version: '{version_from_runscript}' is requested but the model {model} does not have any versions"
+        esm_parser.user_note("Warning", msg)
+        return True
+
+    # exit if requested version is not found
+    if version_from_runscript not in available_versions: 
+        error_type = "Version not available"
+        error_msg = f"Version '{version_from_runscript}' is not available for model {model}"
+        esm_parser.user_error(error_type, error_msg)
+
+    # version requested in the runscript is found in the model configuration
+    return True
+
+
+def check_scenario(config, model_config, runscript, model):
+    """TODO: fill docstring
+    """
+    # model is not specified in the runscript. Using the default
+    if model not in runscript:
+        return True
+    scenario_from_runscript = runscript[model].get("scenario", None)
+
+    # populate list of available scenarios
+    available_scenarios = []
+    items = model_config.get("available_scenarios", None)
+    if items:
+        available_scenarios += items
+    items = model_config.get("choose_scenario", None)
+    if items:
+        available_scenarios += list(items)
+    if available_scenarios:
+        # TODO: I have a better way of handling int/str in versions but this is
+        # for the clean-up phase
+        available_scenarios = [str(item) for item in available_scenarios]
+        available_scenarios = sorted(list(set(available_scenarios)))
+
+    if scenario_from_runscript is None:
+        return True
+    elif available_scenarios is None:
+        msg = f"Scenario: '{scenario_from_runscript}' is requested but the model {model} does not have any scenarios"
+        esm_parser.user_note("Warning", msg)
+        return True
+
+    # exit if requested scenario is not found
+    if scenario_from_runscript not in available_scenarios:
+        error_type = "Scenario not available"
+        error_msg = f"Scenario '{scenario_from_runscript}' is not available for model {model}"
+        esm_parser.user_error(error_type, error_msg)
+
+    # scenario requested in the runscript is found in the model configuration
+    return True
```

Comment:
As I mentioned in our talk, `scenario` is not a ESM-Tools featured variable, meaning that it means nothing to ESM-Tools and should not be handled here, but from the YAML files, with the "yaml warnings/errors". In turn, I agree with the checks of the `version` as it is a featured variable in ESM-Tools.